leef/leef_encoder.go

```diff
@@ -183,40 +183,85 @@ func Encode(msg map[string]interface{}) (string, error) {
 		if !reflect.ValueOf(msg[key]).IsValid() {
 			continue
 		}
+		the_type := reflect.ValueOf(msg[key]).Type()
+		the_kind := the_type.Kind()
 
-		switch reflect.ValueOf(msg[key]).Type().Kind() {
+		log.Infof("%s",the_type)
+		log.Infof("%s",the_kind)
+		log.Infof("%s",msg[key])
 
-		case reflect.Map:
-		case reflect.Array:
-		case reflect.Slice:
-			// if the value is a map, array or slice, then format as JSON
-			t, err := json.Marshal(msg[key])
-			if err != nil {
-				log.Printf("Could not marshal key %s with value %v into JSON: %s, skipping", key, msg[key], err.Error())
-				continue
-			}
-			val = string(t)
 
-		case reflect.String:
-			// make sure to format strings with the appropriate character escaping
-			// also make sure we reflect the "type" and "cb_version" on to the message header, if present
-			if reflect.ValueOf(msg[key]).Type() == jsonNumberType {
-				val = msg[key].(json.Number).String()
+		switch typed_msg_val := msg[key].(type) {
+
+		case map[string] interface{}:
+			log.Info("map handler")
```

Comment:
same here; we can change these to Debug if they still need to be here - otherwise, remove them completely